appeals/web/src/server/appeals/appeal-details/application-submission-date/__tests__/application-submission-date.test.js

```diff
@@ -46,82 +46,112 @@ describe('application-date', () => {
-				skipPrettyPrint: true
-			}).innerHTML;
-
-			expect(unprettifiedErrorSummaryHtml).toContain('There is a problem</h2>');
+		it('should re-render the Date page with an error message if the provided date month is invalid', async () => {
+			const testCases = [
+				{ value: '', expectedError: 'Date month cannot be empty' },
+				{ value: 'a', expectedError: 'Date month must be a number' },
+				{ value: '0', expectedError: 'Date month must be between 1 and 12' },
+				{ value: '13', expectedError: 'Date month must be between 1 and 12' }
+			];
+
+			for (const testCase of testCases) {
+				const invalidData = {
+					'application-submission-date-day': '1',
+					'application-submission-date-month': testCase.value,
+					'application-submission-date-year': '2021'
+				};
+				nock('http://test/')
+					.get(`/appeals/${appealId}/appellant-cases/${appellantCaseId}`)
+					.reply(200, appellantCaseDataNotValidated);
+
+				const response = await request.post(`${baseUrl}/application-date/change`).send(invalidData);
+				expect(response.statusCode).toBe(200);
+
+				const elementInnerHtml = parseHtml(response.text).innerHTML;
+
+				expect(elementInnerHtml).toMatchSnapshot();
+				expect(elementInnerHtml).toContain('Change date application submitted</h1>');
+
+				const unprettifiedErrorSummaryHtml = parseHtml(response.text, {
+					rootElement: '.govuk-error-summary',
+					skipPrettyPrint: true
+				}).innerHTML;
+
+				expect(unprettifiedErrorSummaryHtml).toContain('There is a problem</h2>');
+				expect(unprettifiedErrorSummaryHtml).toContain(`${testCase.expectedError}</a>`);
+			}
 		});
 
-		it('should re-render to applicationSubmissionDate change page if year not valid', async () => {
-			const invalidData = {
-				'application-submission-date-day': '15',
-				'application-submission-date-month': '15',
-				'application-submission-date-year': 'a'
-			};
-
-			nock('http://test/')
-				.get(`/appeals/${appealId}/appellant-cases/${appellantCaseId}`)
-				.reply(200, appellantCaseDataNotValidated);
-
-			const response = await request.post(`${baseUrl}/application-date/change`).send(invalidData);
-
-			const elementInnerHtml = parseHtml(response.text).innerHTML;
-
-			expect(elementInnerHtml).toMatchSnapshot();
-			expect(elementInnerHtml).toContain('Change date application submitted</h1>');
-
-			const unprettifiedErrorSummaryHtml = parseHtml(response.text, {
-				rootElement: '.govuk-error-summary',
-				skipPrettyPrint: true
-			}).innerHTML;
-
-			expect(unprettifiedErrorSummaryHtml).toContain('There is a problem</h2>');
+		it('should re-render the Date page with an error message if the provided date year is invalid', async () => {
+			const testCases = [
+				{ value: '', expectedError: 'Date year cannot be empty' },
+				{ value: 'a', expectedError: 'Date year must be a number' },
+				{ value: '202', expectedError: 'Date year must be 4 digits' },
+				{ value: 3000, expectedError: 'Date must be today or in the past' }
```

Comment:
very minor, and will probably not affect anything, but this should be a string `'3000'` to be consistent with the other test values (and since the real value would be from a text input)

Reply:
done